From cb4d670d8f0f6ef4725094ea18a62729d852ce67 Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Fri, 22 Apr 2022 19:34:52 +0200
Subject: [PATCH 044/127] scripts/glibcelf.py: Mark as UNSUPPORTED on Python
 3.5 and earlier

enum.IntFlag and enum.EnumMeta._missing_ support are not part of
earlier Python versions.

(cherry picked from commit b571f3adffdcbed23f35ea39b0ca43809dbb4f5b)
---
 scripts/glibcelf.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/scripts/glibcelf.py b/scripts/glibcelf.py
index 8f7d0ca184..da0d5380f3 100644
--- a/scripts/glibcelf.py
+++ b/scripts/glibcelf.py
@@ -28,6 +28,12 @@ import collections
 import enum
 import struct
 
+if not hasattr(enum, 'IntFlag'):
+    import sys
+    sys.stdout.write(
+        'warning: glibcelf.py needs Python 3.6 for enum support\n')
+    sys.exit(77)
+
 class _OpenIntEnum(enum.IntEnum):
     """Integer enumeration that supports arbitrary int values."""
     @classmethod
-- 
2.36.1

